### ISSUE TITLE
Pagination doesn't work in L5

### ISSUE THREAD

**Issue by billmn** [nicolaslopezj/searchable#21]
Hi,
I have some problems using Searchable with L5 pagination.

Using "get" all works fine:
```php
User::search('...')->get();
```

But with "paginate" I have an error:
```php
User::search('...')->paginate(10);
```

![schermata 2014-12-28 alle 17 46 39](https://cloud.githubusercontent.com/assets/779534/5564359/daa85170-8eb9-11e4-960d-cc8c649fe81b.png)

The problem seems to be "HAVING" on pagination count ... take a look here: https://github.com/laravel/framework/blob/master/src/Illuminate/Database/Eloquent/Builder.php#L239

Thoughts?

**Comment by woodytang** [comment]
I am basically new to Laravel and have no idea about how it works internally. However, I need to use this package in my latest project desperately because I don't know how to write search function either. 
You guys have no idea how much it took for me  hacking the core of laravel to make it work. I almost cried my eyes out. But, I made it , it works for now, in a STUPID way:

First, for the count(); it is the major issue messing up everything.

I don't and not able to care about how it work, but I find an alternative solution to use it:

      $posts = Post::search($query)->get();

      $counts = count($posts);  //instead of count()


and Next is the painful hacking of Pagination:

1. the paginate() method locates in Eloquent/Builder; there is parameter $total using count() which in this case I will not. so I passed the $counts to paginate() method instead of its internal computing.

2. this cause a serials of core extending(hacking)
I extended:
a. Model;
b. QueryBuilder,
c. EloquentBuilder
d. LengthAwarePaginator

and made my Post Model extend the custom Model.

It really took me a while to figure out which is which;

3.  in my custom EloquentBuilder, I create a new paginate() function named search_paginate() so that it won't conflict with those normal paginators.

4. I passed $counts to search_paginate() to replace its original functionality, and then to CustomLengthAwarePaginator , and it worked!!

5. But there is an issue, everytime i click the pages in pagination buttions, it will fire a new request of query, which apparently is null, so the webpage populates all the posts;

6. I guess I need to make paginator to remember the first query,, so I managed to pass the query parameter  a long way to url() method in CustomLengthAwarePaginator.

and it finally worked..

trust me this is a pain.

@nicolaslopezj  pleas have a look into that specific issue, hopes the above will help, your efforts will make this world a better place, thank you!
